### PR TITLE
routing: add all optional parameters

### PR DESCRIPTION
**Pull request by rayanosseiran**
Please review [PR#3](https://github.com/TeamW-P/RNABayesPairing2/pull/3) before reviewing this.

This series of commits is focused on implementing functional BayesPairing routing. Currently, it covers:

- Standard calls to BayesPairing including string & file support.
- Implements a new class, pipeline_bp, in-order to have BayesPairing return output directly instead of saving it to a file. Inputs are stored locally through the use of Python's tempfile module which allows safe unique temporary storage and permanent destruction once complete.
- Implements pipeline_chefschoice which generates an SVG for output.
- Both string and file input are handled via a POST and as a result, data is provided via the request body. String input is handled via POST for the sake of consistency.
- Implements representative graph generation which takes a module or a list of modules. Searches the default dataset.  Data produced by this method represents what is passed to VeRNAl/RNAMigos. A parameter "get_graphs" is used to indicate whether this should be run within BayesPairing calls. The frontend team will also call this endpoint through the pipeline.
- Switches to the new version of BayesPairing. 

BayesPairing is completely functional. The only work that remains is clean up and improved error handling (see issue #6).

### CODE REVIEW

**Review by ajaypatel24 — APPROVED** [review]
looks good

File: routes/route.py [hunk 1]
```diff
@@ -1,58 +1,291 @@
...
+        arguments["w"] = input.get("window_length", default=200, type=int)
+        arguments["s"] = input.get("step_size", default=100, type=int)
+        if (input.get("modules")):
+            arguments["mod"] = eval(input.get("modules"))
+        else:
+            arguments["mod"] = None
+        arguments["constraints"] = input.get(
+            "constraints", default="", type=str)
+        arguments["aln"] = input.get("alignment_provided", default=0, type=int)
+        arguments["verbose"] = BAYESPAIRING_VERBOSE
```
Comment (larakollokian) [inline routes/route.py:168]: is there an easier way to do this, like all the arguments at once instead of one at a time?

**Review by larakollokian — APPROVED** [review]
From my limited understanding of BP2, these changes look good. However, I do have some general comments:

1. For the next PRs, it would be nice if you could separate your refactors of Roman's code and your actual changes into different PRs. It would make the whole thing easier to digest and focus on. The way you have it now, it's not obvious what changes you made vs Roman made (especially all the changes of directory paths, etc).
2. Too many changes in one PR. "Switches to the new version of BayesPairing" should have been its own thing imo because again I wasn't sure what parts to focus on while reviewing.

But mostly looking at `routes.py`, everything looks clean. I agree that POST makes more sense in this context. I also agree that we need to figure out error handling.

Also idk if this is really extra but the helper functions you have in `routes.py` aren't directly routing functions (like `bayespairing` and `retrieve_graphs`), so I'm thinking maybe they shouldn't be in that file? Not sure if that would make things more complicated to follow though, so lemme know your thoughts.

**Comment by larakollokian** [comment]
also why are you merging into experimental branch rather than main?

File: routes/route.py (hunk 1 above)
Comment (rayanosseiran) [reply routes/route.py:168]: Cleaned up in follow-up.

**Pull request merged**